File: cpp/kode program/MenuItem.cpp
#ifndef MENUITEM_CPP
#define MENUITEM_CPP

#include <iostream>
#include <string>
using namespace std;

// kelas menuitem sebagai parent untuk food dan drink
class MenuItem {
protected:
    int id;         // menyimpan id item
    string name;    // menyimpan nama item
    int price;      // menyimpan harga item

public:
    MenuItem() {}

    // setter dan getter untuk atribut id
    void setId(int id) { 
        this->id = id; 
    }
    int getId() { 
        return id; 
    }

    // setter dan getter untuk atribut name
    void setName(string name) { 
        this->name = name; 
    }
    string getName() { 
        return name; 
    }

    // setter dan getter untuk atribut price
    void setPrice(int price) { 
        this->price = price; 
    }
    int getPrice() { 
        return price; 
    }

    // method virtual untuk menampilkan informasi item
    // bisa dioverride di class turunan (polymorphism)
    virtual void printInfo() {
        cout << "ID: " << id
             << " | Name: " << name
             << " | Price: " << price;
    }
};

#endif

File: python/kode program/MenuItem.py
class MenuItem:
    def __init__(self, item_id=0, name="", price=0):
        self._id = item_id      # atribut id (protected)
        self._name = name       # atribut name (protected)
        self._price = price     # atribut price (protected)

    # setter & getter untuk id
    def set_id(self, item_id):
        self._id = item_id

    def get_id(self):
        return self._id
    
    # setter & getter untuk name
    def set_name(self, name):
        self._name = name

    def get_name(self):
        return self._name

    # setter & getter untuk price
    def set_price(self, price):
        self._price = price

    def get_price(self):
        return self._price

    # method polymorphism untuk menampilkan info dasar
    def base_info(self):
        return f"id: {self._id} | name: {self._name} | price: {self._price}"
